src/v/cluster/cluster_discovery.cc

```diff
@@ -212,7 +208,7 @@ cluster_discovery::request_cluster_bootstrap_info_single(
               !*_is_cluster_founder,
               "We can only detect the presence of a cluster (indicating we are "
               "not a founder) early, not the absence");
-            co_return reply_result.value();
+            co_return cluster_bootstrap_info_reply{};
```

Comment:
I understand that this value won't be used if returned, but we're not really saving anything returning an empty reply, and it takes some effort to prove it's not used, making it difficult to reason about it.

Reply:
This is actually important so we don't throw, since there is no value if the result is initialized to an error.